orc/src/main/java/org/apache/iceberg/orc/ORCSchemaUtil.java

```diff
@@ -269,7 +269,7 @@ private static TypeDescription buildOrcProjection(Integer fieldId, Type type, bo
         break;
       case MAP:
         Types.MapType map = (Types.MapType) type;
-        TypeDescription keyType = buildOrcProjection(map.keyId(), map.keyType(), true, mapping);
```

Comment:
Why was this change required?

Reply:
While a `MapType ` column is optional if insert a  empty map like `null`,  the `key` and `value`  both are `null`

Reply:
Map keys are always required. Does this make the map key optional?